index.js

```diff
@@ -2372,24 +2372,31 @@ var HyperHTMLElement = (function (exports) {
       if (options && options.extends) {
         const Native = document.createElement(options.extends).constructor;
         const Intermediate = class extends Native {};
-        const Super = getPrototypeOf(Class);
-        ownKeys(Super)
-          .filter(key => [
-            'length', 'name', 'arguments', 'caller', 'prototype'
-          ].indexOf(key) < 0)
-          .forEach(key => defineProperty(
-            Intermediate,
-            key,
-            getOwnPropertyDescriptor(Super, key)
-          )
-        );
-        ownKeys(Super.prototype)
-          .forEach(key => defineProperty(
-            Intermediate.prototype,
-            key,
-            getOwnPropertyDescriptor(Super.prototype, key)
-          )
-        );
+        let Super, BaseClass = Class;
+        while(Super = getPrototypeOf(BaseClass)){
+          ownKeys(Super)
+            .filter(key => [
+              'length', 'name', 'arguments', 'caller', 'prototype'
+            ].indexOf(key) < 0)
+            .forEach(key => defineProperty(
```

Comment:
as you go top down, you don't want to define overwrites so that the filter should also check that the property doesn't exist already in the `Intermediate` class. An easy way to do this, is to define the keys to filter outside the loop and add each key that wasn't there already.

```js
// example ...
const keys = ['length', 'name', 'arguments', 'caller', 'prototype'];
while(Super = getPrototypeOf(BaseClass)) {
  ownKeys(Super)
    .filter(key => keys.indexOf(key) < 0)
    .forEach(key => {
      keys.push(key);
      defineProperty(Intermediate, key, getOwnPropertyDescriptor(Super, key));
    })
}
```

In this way, if `B` has `method` and it was overwriting the `A` `method`, the `Intermediate` would get `B` and it will ignore `A`, I hope it's clear

Reply:
Very clear, thanks. I had to add it to both loops, so now that both loops do the same but for different targets, I simplified it into one loop.

Let me know what you think.

Thanks!